### PR TITLE
#100 - finished PhotoInput

### PR DESCRIPTION
**Pull request by boltonlin**
[Ticket](https://trello.com/c/E9R1thPI/100-allow-for-upload-of-photos-for-add-answer-modal)
- user can now upload up to 5 photos with answers
- implements front-to-back request to [imgbb api](https://api.imgbb.com/) for image hosting
- fully works with `QandAModal.jsx`

### CODE REVIEW

File: client/src/components/Questions/PhotoInput.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    const copy = [...files];
+    copy.splice(i, 1);
+    setFiles(copy);
+  };
+
+  return (
+    <div className="qa add-photos">
+      {files.map((file, i) => (
+        <div
+          // Exception: Items will never change order, no valid ID.
```
Comment (eriknewland) [inline client/src/components/Questions/PhotoInput.jsx:40]: Want to clarify if backing out of uploading the file will break this code. I noticed that combining input type="file" and onChange could lead to this bug. The solution seems to be use onLoad or ensure e.target.files[0] exists prior to mapping the thumbnail

File: client/src/components/Questions/QandAModal.jsx [hunk 2]
```diff
@@ -74,18 +76,38 @@ export default function QandAModal({
...
+
   const submitForm = (e) => {
     e.preventDefault();
-    if ((e.type === 'click' || e.key === 'Enter') && validateForm()) {
+    if ((e.type === 'click' || e.key === 'Enter')) {
+      // TODO: make custom alert message
+      if (!validateForm()) {
+        let message = '';
+        if (!isValid('body') || !isValid('name')) message += 'Please fill the following:\n';
+        if (!isValid('body')) message += ` - ${type.charAt(0).toUpperCase()}${type.slice(1)}\n`;
```
Comment (eriknewland) [inline client/src/components/Questions/QandAModal.jsx:92]: Good UX for for handling mandatory fields.

File: server/index.js [hunk 3]
```diff
@@ -10,9 +10,7 @@ const app = express();
 
 /* MIDDLEWARE */
 app.use(morgan('dev'));
-app.use(express.json({
-  limit: '20mb',
-}));
+app.use(express.json({ limit: '32mb' }));
```
Comment (eriknewland) [inline server/index.js:13]: is 32mb max size for imgbb?

**Review by eriknewland — COMMENTED** [review]
Couple of comments we can discuss but no major issues.

File: server/index.js (hunk 3 above)
Comment (boltonlin) [reply server/index.js:13]: [yes ](https://api.imgbb.com/)
image

File: client/src/components/Questions/PhotoInput.jsx (hunk 1 above)
Comment (eriknewland) [reply client/src/components/Questions/PhotoInput.jsx:40]: Demo'd working fix

**Review by eriknewland — APPROVED** [review]
Edge-case bug resolved

**Pull request merged**